### DISCUSSION TITLE
Cookie domain name

### DISCUSSION THREAD

**Discussion by ilamarche** [directus/directus#3614] [Feature Requests]
Hello,

Within the framework of a project, I created a three-tier architecture with one domain and two sub-domains:

- api.xxxxxx.local
- admin.xxxxxx.local
- xxxxxx.local

I have noticed that when I create a separate frontend, the domain name associated with the user session cookie is not the correct one.
It could be interesting to add an environment variable allowing to manage sessions on the same domain, for example :
REFRESH_TOKEN_COOKIE_DOMAINE='.xxxxxx.local''.

https://github.com/directus/directus/blob/main/api/src/controllers/auth.ts

**Comment by rijkvanzanten** [comment]
What do you mean by

>  the domain name associated with the user session cookie is not the correct one.

What is it, what did you expect/need it to be? Right now, the cookie is/should be set under the api.xxxx.local domain, as that's where you log in which (afaict) is correct 🤔

**Reply by ilamarche** [reply]
For domain names, the idea is to have my client site on _exemple.local_, the administration area of directus on _admin.exemple.local_ and the api directus on _api.exemple.local_.

I would like to share the user session if necessary between the client site and the administration area by putting in the cookie the domain name: _.example.local_

However, when I connect my user from my client site, the domain name set in the cookie is _api.example.local_ and not _example.local_, what will not allow the user's session to be maintained.

**Reply by rijkvanzanten** [reply]
The Directus admin is served by the API though, so how are you running the app under a different domain? 🤔 

> However, when I connect my user from my client site, the domain name set in the cookie is api.example.local and not example.local, what will not allow the user's session to be maintained.

Makes sense to me 👍🏻 I'm down to keep discussing an implementation of an env var for this, but I want to make 100% sure that there's no unexpected CORS-side-effects that people will run into. Browsers are getting more and more aggressive when it comes to third-party cookies, and while in your case it's the same root domain, it's trivially easy to accidentally break your API by setting that value to something nonsensical 🤔

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
